R/plot_fns.R

```diff
@@ -4128,304 +4164,251 @@ plot.rmdFilt <- function(x, pvalue_threshold = NULL, sampleID = NULL,
-      if (sum(goodies_alpha) > 0) {
-        # Snatch the sample name and pair name as they will be used in multiple
-        # places. It will save some typing. ... However, all the typing I just
-        # saved has probably been undone by writing this comment.
-        sample_name <- attr(filter_object, "fdata_cname")
-        pair_name <- attr(attr(filter_object, "group_DF"), "pair_id")
-
-        # Grab the names of filtered samples.
-        filtad <- as.character(filter_object[goodies_alpha, sample_name])
-
-        # !#!#!#!#!#!#!#!#!#!
-        # The following code checks if the samples in a pair will be split. For
-        # example, one sample in a pair will be filtered and another sample in
-        # the pair will not be filtered. If a pair is split remove ALL samples
-        # associated with that pair.
-        # !#!#!#!#!#!#!#!#!#!
-
-        # Snag the associated pair IDs for the samples that will be filtered.
-        filtad_pairs <- attr(filter_object, "fdata") %>%
-          dplyr::filter(!!dplyr::sym(sample_name) %in% filtad) %>%
-          dplyr::pull(!!dplyr::sym(pair_name))
-
-        # Go back to f_data and nab all the sample names corresponding to the
-        # pair IDs associated with the original samples that were selected for
-        # removal. These sample names will be used to change the point shape.
-        # The points that will be different are the ones corresponding to the
-        # samples that belong to pair where only one sample falls below the
-        # threshold.
-        filtad_too <- attr(filter_object, "fdata") %>%
-          dplyr::filter(!!dplyr::sym(pair_name) %in% filtad_pairs) %>%
-          dplyr::pull(!!dplyr::sym(sample_name)) %>%
-          as.character()
-
-        # Update the goodies_alpha vector to reflect the additional samples that
-        # will be filtered.
-        goodies_alpha <- filter_object[, sample_name] %in% filtad_too
-
-        # Create a vector of samples who are guilty by association. (They do not
-        # fall below the threshold but their partners do.)
-        condemned <- setdiff(filtad_too, filtad)
-
-        # Create a logical vector that will determine the point shape.
-        goodies_pch <- filter_object[, sample_name] %in% condemned
-
-      } else goodies_pch <- rep(FALSE, nrow(filter_object))
-
+    
+    ## Alpha ##
+    if(!is.null(pvalue_threshold)){
+      
+      # Determine which samples fall below the p-value threshold (outliers)
+      filter_object$out <- filter_object$pvalue < pvalue_threshold
+      
+      ## Define plotting features
+      
+      # Shape and Alpha (opacity) #
+      if (!is.null(attr(group_df, "pair_id"))) {
+        
+        pair_id <- attr(group_df, "pair_id")
+        
+        ## Get pairs that have divergence in outlier status
+        pair_df <- filter_object %>% 
+          dplyr::group_by_at(dplyr::vars(pair_id)) %>% 
+          dplyr::reframe(
+            diverge = any(out) && !all(out)
```

Comment:
I believe to get rid of the 'no visible binding for global variable...' NOTE, you can replace with .data$out

Reply:
@clabornd That should be what this does since out is a column (reframe is similar to summarize in that way) -- also I don't get this error, but it seems to be triggered if a global variable is set to "out"? can you retry this and send me a screenshot if it pops up again?

Reply:
I found the linky to the fix:
https://github.com/STAT545-UBC/Discussion/issues/451#issuecomment-264598618

Though I'm still mystified as to why it can't seem to handle it in the test environment specifically, so strange

Reply:
Yeah I think you're correct, different test functions trip different behavior from what I can tell:

<img width="839" alt="image" src="https://github.com/user-attachments/assets/da99b7de-b609-4f5f-8dcb-3f91cb8723a3" />

e.g. devtools here will trip different warnings